src/movies.js

```diff
@@ -1,25 +1,140 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const mapMovies = moviesArray.map(movie => movie.director);
+    return mapMovies;
+}
+//console.log(getAllDirectors(movies));
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    if (!moviesArray.length) return 0;
+    const spielbergMovies = moviesArray.filter(movie => movie.director === 'Steven Spielberg' && movie.genre.includes('Drama'));
+    //console.log(spielbergMovies);
+    return spielbergMovies.length;
+}
+//console.log(howManyMovies(movies))
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (!moviesArray.length) return 0;
+    
+    const sumScores = moviesArray.reduce((acc, movie) => {
+        if (movie.score) return acc + movie.score;
+        else return acc;
+    }, 0)
+    const averageScores = (sumScores / moviesArray.length);
+    return Math.round(averageScores* 100)  / 100;
+}
+//console.log(scoresAverage(movies));
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+    const getDrama = moviesArray.filter(movie => movie.genre.includes('Drama'))
+    if (!getDrama.length) return 0
+    const sumScores = getDrama.reduce((acc, movie) => {
+        if (movie.score) return acc + movie.score;
+        else return acc;
+}, 0);
+    const averageScores = (sumScores / getDrama.length);
+    return Math.round(averageScores * 100)  / 100;
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+   /*  const titleSort = moviesArray.title.sort() */
+    const movArrCopy = moviesArray.map((x) => x)
+    const yearSort = movArrCopy.sort((a, b) => a.year - b.year);
+    return yearSort;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+    const movieTitles = moviesArray.map(movie => movie.title);
+    const alphaSort = movieTitles.sort();
```

Comment:
After you clone the original array and grab the titles, you could just do the `.sort()` and `.slice()` methods at once `const alphaSort = movieTitles.sort().slice(0, 20);` 
That saves lines of code and prevents errors in the `.filter()`